packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -87,6 +87,16 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
             userContentController: WKUserContentController(),
           ),
     );
+
+    webView.uiDelegate = uiDelegate;
+    uiDelegate.setOnCreateWebView((
+      WKWebViewConfiguration configuration,
+      WKNavigationAction navigationAction,
+    ) {
+      if (!navigationAction.targetFrame.isMainFrame) {
+        webView.loadRequest(navigationAction.request);
+      }
+    });
   }
```

Comment:
Relevant code from current implementation:
https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FlutterWebView.m#L699
https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FlutterWebView.m#L121

Reply:
Weird. I'm surprised nobody has ever filed a bug about this behavior, because it's pretty surprising.